ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -424,7 +452,26 @@ int getNumberOfItems() {
   }
 
   /**
-   * Writes the index to a persistent file. Writes the data in the following format
+   * Writes the index to a persistent file. Writes the data in the following format in version 1
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   * | version | keysize | valuesize | fileendpointer |  last modified time | Reset key | Reset key type
+   * |(2 bytes)|(4 bytes)| (4 bytes) |    (8 bytes)   |     (4 bytes)       | (n bytes) |   ( 2 bytes)
+   *  - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *                                                   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *                                                     key 1  | value 1  |  ...  |   key n   | value n   | crc      |
+   *                                                   (n bytes)| (n bytes)|       | (n bytes) | (n bytes) | (8 bytes)|
+   *                                                   - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+   *  version            - the index format version
+   *  keysize            - the size of the key in this index segment
+   *  valuesize          - the size of the value in this index segment
+   *  fileendpointer     - the log end pointer that pertains to the index being persisted
+   *  last modified time - the last modified time of the index segment
+   *  reset key          - the reset key of the index segment
```

Comment:
Will sync up with you offline. This is to track whether the store key is referring to a PUT record or a DELETE record